unitTests/test_slice.cpp

```diff
@@ -117,16 +120,21 @@ TYPED_TEST_P(slice, atAccess)
     }
 }
 
-// TODO C++11: test range based for loop
 TYPED_TEST_P(slice, iteratorAccess)
 {
     Slice<TypeParam> sl = this->getTestSlice();
```

Comment:
[suggestion] Rename `sl` to `slice`. It is important to make the tests as readable as possible. 

Also in line 128, you could use `const auto&` instead of specifying the full type name, right?

And a question: the reason for having these 2 loops is just for demonstrating we can iterate over elements in the two ways right?

Reply:
> [suggestion] Rename `sl` to `slice`. It is important to make the tests as readable as possible.

Can't call this one `slice` since that name is already taken by the class fixture (which I wanted to call `sliceFixture`, as that's what that is, but then all the tests have stupid names).

> Also in line 128, you could use `const auto&` instead of specifying the full type name, right?

Yes, but I wanted this one to be explicit, to catch any type mistakes.

> And a question: the reason for having these 2 loops is just for demonstrating we can iterate over elements in the two ways right?

Yes

Reply:
A convention I used in the past for naming tests with GTest is:

```
TYPED_TEST_P(ASlice, accessElementsWithIterators)
TYPED_TEST_P(ASlice, accessElementsWithAt)
TYPED_TEST_P(ASlice, throwsWhen...)
```
Note that this could be read more or less like normal text once you run the tests:

```
ASlice/accessElementsWithIterators --> A Slice access elements with iterators
ASlice/throwsWhen                  --> A Slice throws when ...
```

This convention would solve your situation. But again, I am just talking about previous experiences. If you do not like them, it is totally fine 😉